packaging/docker/create_server_environment.bash

```diff
@@ -0,0 +1,47 @@
+#! /bin/bash
+
+#
+# create_server_environment.bash
+#
+# This source file is part of the FoundationDB open source project
+#
+# Copyright 2013-2018 Apple Inc. and the FoundationDB project authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+fdb_dir=/var/fdb
+env_file=$fdb_dir/.fdbenv
+
+: > $env_file
+
+source /var/fdb/scripts/create_cluster_file.bash
+
+if [[ "$FDB_NETWORKING_MODE" == "host" ]]; then
+	public_ip=127.0.0.1
+	FDB_CLUSTER_FILE_CONTENTS="docker:docker@$public_ip:4500"
+elif [[ "$FDB_NETWORKING_MODE" == "container" ]]; then
+	public_ip=$(grep `hostname` /etc/hosts | sed -e "s/\s *`hostname`.*//")
```

Comment:
Should the `FDB_CLUSTER_FILE_CONTENTS` be set here as well? I tried:

```
docker build . -t fdb:latest --build-arg FDB_VERSION=5.2.5
docker run -ti --rm -e "FDB_NETWORKING_MODE=container" -e "FDB_PORT=4500" fdb:latest
```

I got:

```
Starting FDB server on 172.17.0.2:4500
Invalid cluster file `/var/fdb/fdb.cluster': 2104 Connection string invalid
Error: Connection string invalid
```

Adding the `FDB_CLUSTER_FILE_CONTENTS` line 32 in this elsif block seemed to make things better but I'm not too familiar with FDB to know if this should have been written out by default and should "just work" or my setup is just wrong.

Reply:
Yes, good catch! I think I was trying to accomplish this through the FDB_COORDINATOR option, but the DNS resolution wasn't working properly, so I've tweaked this to use FDB_CLUSTER_FILE_CONTENTS in both networking modes.